### PR TITLE
WIP: add support for .NET Core compilation

### PR DESCRIPTION
**Pull request by lahma**
- add shims for missing features on .NET Core
- add xproj-based projects and solution, will be migrated later with VS'15
- remove packages.config
- update .gitignore

### CODE REVIEW

**Comment by lahma** [comment]
I've done some base work to support .NET Core. There are some questions that need to be addressed before this can be considered more mature though:
- how are builds made? at the moment dotnet tooling supports packaging using "dotnet pack -c Release" and can be run in main DLL project folder "PhysicalMeasure"
  - this requires still additional settings to project.json to add metadata to nuget packages
- what will be the .NET Framework target version, I see that 4.6 is now set in project properties but the lib could easily target lower versions
- do you need build scripts? I could create simple FAKE build script that produces artifacts and runs tests

In short, with this branch you can already create netstandard1.0 packages which can run basically anywhere.
